### DISCUSSION TITLE
How do I interpret the raw data from the monitor for IR codes?

### DISCUSSION THREAD

**Discussion by gso222** [Arduino-IRremote/Arduino-IRremote#1212] [Q&A] [answered]
End goal: Control Fujitsu Halcyon minisplit with Uno R4 Wifi.

Successes: I have implemented the ReceiveAndSend example. I start the code, point my working remote at my IR receiver, and send an "Off" command. I then allow my device to see my IR LED, and push a button connected to D5. The example code repeats out my "Off" command. The Fujitsu successfully receive the command and turns off!

Problem 1: I don't understand how to use the data sent to the serial monitor to allow me to build a send-only code that sends this command at the push of a button. The sender examples are looking for commands and addresses. 

First is the Leader code of +3300,-1650. Then I have +4xx followed by a -4xx for a logical False, or +4xx followed by a -12xx for a logical True. This gives me `00101000110001100000000000001000000010000100000010111111` for my address and command. Can I assume it doesn't matter what is address and what is command? `Solved: ReceiveDemo gave me the code needed to proceed, bypassing the roadblock of me understanding what this binary signal translates to.`

```
16:27:37.551 -> Received unknown code and store 115 timing entries as raw
16:27:37.551 -> rawData[116]: 
16:27:37.551 ->  -1030700
16:27:37.551 ->  +3300,-1650
16:27:37.551 ->  + 400,- 400 + 450,- 400 + 400,-1250 + 400,- 400
16:27:37.551 ->  + 450,-1200 + 450,- 400 + 400,- 400 + 450,- 400
16:27:37.551 ->  + 400,-1200 + 450,-1200 + 450,- 400 + 400,- 400
16:27:37.593 ->  + 450,- 400 + 400,-1250 + 400,-1250 + 400,- 400
16:27:37.593 ->  + 450,- 400 + 450,- 350 + 450,- 400 + 400,- 400
16:27:37.593 ->  + 450,- 400 + 400,- 400 + 450,- 400 + 400,- 400
16:27:37.593 ->  + 450,- 400 + 400,- 400 + 450,- 400 + 400,- 400
16:27:37.593 ->  + 450,-1200 + 450,- 400 + 400,- 400 + 450,- 400
16:27:37.593 ->  + 400,- 400 + 450,- 400 + 400,- 400 + 450,- 400
16:27:37.593 ->  + 400,-1250 + 400,- 400 + 450,- 400 + 400,- 400
16:27:37.593 ->  + 450,- 400 + 400,-1250 + 400,- 400 + 450,- 400
16:27:37.593 ->  + 400,- 400 + 450,- 400 + 400,- 400 + 450,- 400
16:27:37.593 ->  + 400,-1250 + 400,- 400 + 450,-1200 + 450,-1200
16:27:37.628 ->  + 400,-1250 + 400,-1250 + 400,-1200 + 450,-1200
16:27:37.628 ->  + 450
16:27:37.628 -> Sum: 64700`
```
Problem 2 to solve another day: Most commands (e.g. "On") have a length of 259, and this example code can't handle it. 255 or 256 appears to be the maximum length. `Solved: The ReceiveAndSend may not be able to handle this length, but ReceiveDemo got it all.`
```
16:40:31.973 -> Received unknown code and store 259 timing entries as raw 
16:40:31.973 -> rawData[4]: 
16:40:31.973 ->  -3276750
16:40:31.973 ->  +3350,-1550
16:40:31.973 ->  + 450
16:40:31.973 -> Sum: 5350
```

**Comment by ArminJo** [comment, marked as answer]
Use the **ReceiveDemo example** to print out all informations about your IR protocol.

**Reply by gso222** [reply]
Thank you, it is working perfectly now. I ran and Off and On command on ReceiveDemo which resulted in the following.

```
17:26:49.057 -> Protocol=PulseDistance Raw-Data=0xFD021010006314 56 bits LSB first
17:26:49.057 -> Send on a 32 bit platform with: IrSender.sendPulseDistanceWidth(38, 3250, 1650, 400, 1200, 400, 400, 0xFD021010006314, 56, PROTOCOL_IS_LSB_FIRST, <RepeatPeriodMillis>, <numberOfRepeats>);
17:26:52.526 -> 
17:26:52.526 -> Protocol=PulseDistance Raw-Data=0x2B20000000000481 128 bits LSB first
17:26:52.526 -> Send on a 32 bit platform with: 
17:26:52.526 ->     uint64_t tRawData[]={0x3009FE1010006314, 0x2B20000000000481};
17:26:52.573 ->     IrSender.sendPulseDistanceWidthFromArray(38, 3300, 1650, 450, 1200, 450, 400, &tRawData[0], 128, PROTOCOL_IS_LSB_FIRST, <RepeatPeriodMillis>, <numberOfRepeats>);
```
